Source/missiles.cpp

```diff
@@ -2340,12 +2340,8 @@ void AddHeal(Missile &missile, const AddMissileParameter & /*parameter*/)
 	auto &player = Players[missile._misource];
 
 	int hp = (GenerateRnd(10) + 1) << 6;
-	for (int i = 0; i < player._pLevel; i++) {
-		hp += (GenerateRnd(4) + 1) << 6;
-	}
-	for (int i = 0; i < missile._mispllvl; i++) {
-		hp += (GenerateRnd(6) + 1) << 6;
-	}
+	hp += GenerateRndSum(4, player._pLevel) + player._pLevel << 6;
+	hp += GenerateRndSum(6, missile._mispllvl) + missile._mispllvl << 6;

```

Comment:
All other cases of mixed addition and bitwise shift operators uses parentheses to explicitly define the order of precedence.

```suggestion
	hp += (GenerateRndSum(4, player._pLevel) + player._pLevel) << 6;
	hp += (GenerateRndSum(6, missile._mispllvl) + missile._mispllvl) << 6;
```

Although, it might also be even better to run the computations first and do one shift operation at the end.

```c++
	int hp = GenerateRnd(10) + 1;
	hp += GenerateRndSum(4, player._pLevel) + player._pLevel;
	hp += GenerateRndSum(6, missile._mispllvl) + missile._mispllvl;
	hp <<= 6;
```

Reply:
Thank you!  I changed it to:
	int hp = GenerateRnd(10) + 1;
	hp += GenerateRndSum(4, player._pLevel) + player._pLevel;
	hp += GenerateRndSum(6, missile._mispllvl) + missile._mispllvl;
	hp <<= 6;